src/rules/import-alias.ts

```diff
@@ -10,23 +10,72 @@ import type { JSONSchema4 } from "json-schema";
 import { dirname, join as joinPath, resolve, sep as pathSep } from "path";
 import slash from "slash";
 
+interface ConfigMatch {
+    type: "pattern" | "path";
+    length: number;
+    depth: number;
+}
+
 function isPermittedRelativeImport(
     importModuleName: string,
     relativeImportOverrides: RelativeImportConfig[],
-    fileAbsoluteDir: string
+    fileAbsoluteDir: string,
+    filepath: string
 ) {
-    const configOfPath = relativeImportOverrides.find((config) =>
-        fileAbsoluteDir.includes(resolve(config.path))
-    );
-    if (!configOfPath) {
-        return false;
-    }
-
     const importParts = importModuleName.split("/");
     const relativeDepth = importParts.filter(
         (moduleNamePart) => moduleNamePart === ".."
     ).length;
-    return configOfPath.depth >= relativeDepth;
+
+    const pathsMatchs: ConfigMatch[] = [];
+    const patternsMatchs: ConfigMatch[] = [];
```

Comment:
While your suggestion is fairly reasonable, it feels like it could lead to unexpected behavior in some situations. Let's keep it simple and just prefer the highest depth configuration that matches.

Reply:
yeah i'll switch on sorting by depth